packages/teleport/src/Nodes/AddNode/Automatically/Automatically.tsx

```diff
@@ -14,15 +14,29 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import { State } from './../useAddNode';
+import React, { useEffect } from 'react';
 import TextSelectCopy from 'teleport/components/TextSelectCopy';
-import { Alert, Text, Indicator, Box, ButtonLink } from 'design';
+import {
+  Alert,
+  Text,
+  Indicator,
+  Box,
+  ButtonLink,
+  ButtonSecondary,
+} from 'design';
+import { DialogContent, DialogFooter } from 'design/Dialog';
+import { createBashCommand, State } from './../useAddNode';
 
 export default function Automatically(props: Props) {
-  const { script, expiry, createJoinToken, attempt } = props;
+  const { createJoinToken, attempt, onClose, joinToken } = props;
 
-  if (attempt.status === 'processing') {
+  useEffect(() => {
+    if (!joinToken) {
+      createJoinToken();
+    }
+  }, [joinToken]);
```

Comment:
hm, do we need to state `joinToken` as a dependency?

Reply:
Not really. Some lint rules (that we don't use) would complain if we didn't add `joinToken` here. But I think in this case it is ok to remove it